### DISCUSSION TITLE
Does rspack have a max file path name size for compiling node modules?

### DISCUSSION THREAD

**Discussion by ghost** [web-infra-dev/rspack#6966] [Q&A] [answered]
Not sure if this is a bug or a [side effect of Rust](https://github.com/rust-lang/cargo/issues/9770) but I'm encountering a compilation error and I'm 99% sure it's because the file path of the module index file is too long. I'm in the process of converting a react (18.2.0) webpack project to rspack and when I try to build it, I'm getting the following error in the console:

```
ERROR in provide shared module (default) {file path}
  × Resolve error: Can't resolve  {file path}
   ╭────
   ╰────

Rspack 0.7.4 compiled with 6 errors in 16.53 s
```

Since this is a company project, I don't want to show the full file path, but it's 297 characters. Since it's locked to source control, moving the project closer to the root is not an option. Here's the relevant part of the file path with my company name excluded:

\\\\?\C:\\...\ClientApp\node_modules\\.pnpm\\@\*\*\*\*\*\*\*\*+use-federated-component@1.1.0_react-dom@18.2.0_react@18.2.0__react@18.2.0_webpack@5.92.1_esbuild@0.20.2_\node_modules\@\*\*\*\*\*\*\*\*\use-federated-component\dist\esm\index.js

I'm wondering if there's a max file path size for rspack to compile? Does rspack have a method to allow these to compile or am I just out of luck? 

Edit: I should note the reason I think this is a file path issue is because there were two errors, but I shortened the branch name and that got rid of one of the errors because the file path went below 255 characters.

**Comment by h-a-n-a** [comment, marked as answer]
Woud you please add `node-linker=hoisted` to `.npmrc` to see if the problem dismissed?

**Reply by ghost** [reply]
Looks like that fixed it! Thank you so much

**Reply by h-a-n-a** [reply]
This is not an ideal way to fix that tho. This downgrades pnpm's isolated dependency to a flat dependency map.
